node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -194,6 +211,8 @@ struct PerRequest {
 	to_requester: oneshot::Sender<(CandidateReceipt, PoV)>,
 	/// A jaeger span corresponding to the lifetime of the request.
 	span: Option<jaeger::Span>,
+	/// A metric histogram for the lifetime of the request
+	_lifetime_timer: Option<metrics::prometheus::prometheus::HistogramTimer>,
```

Comment:
```suggestion
	maybe_span_timer: Option<metrics::prometheus::prometheus::HistogramTimer>,
```

Reply:
I agree that this would be the correct approach, it's not an unused field after all

Reply:
Well, this will cause compile warning, as the only usage of this field is an implicit observe on drop. On the contrary jaeger `span` field is actually used in the code explicitly.

Reply:
I can rename `_lifetime_timer` to `_maybe_span_timer` of course but I'm not sure if that will improve readability.